legend-pure-core/legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/serialization/grammar/m3parser/antlr/AntlrContextToM3CoreInstance.java

```diff
@@ -1131,89 +1138,142 @@ else if (oneColSpec.type() != null)
 
             // FunctionName
             boolean isArray = ctx.columnBuilders().BRACKET_OPEN() != null;
-            List<Boolean> nonFunctions = ListIterate.collect(ctx.columnBuilders().oneColSpec(), x -> x.type() != null | x.COLON() == null).distinct();
-            if (isArray && nonFunctions.size() > 1)
+            List<RelationColumnType> relationColumnTypes = ListIterate.collect(ctx.columnBuilders().oneColSpec(), col ->
```

Comment:
This change is to improve the readability of the code, I think it's better if we clearly show what we do for each scenario (i.e. column definition, aggregate, extend), if you disagree, I can revert this part.